pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -748,22 +767,24 @@ func (c *Reconciler) runNextSchedulableTask(ctx context.Context, pr *v1beta1.Pip
 		}
 	}
 
+	// Validate parameter types in matrix after apply substitutions from Task Results
 	for _, rpt := range nextRpts {
-		if rpt.IsFinalTask(pipelineRunFacts) {
-			c.setFinallyStartedTimeIfNeeded(pr, pipelineRunFacts)
-		}
-		if rpt == nil || rpt.Skip(pipelineRunFacts).IsSkipped || rpt.IsFinallySkipped(pipelineRunFacts).IsSkipped {
-			continue
-		}
-
-		// Validate parameter types in matrix after apply substitutions from Task Results
 		if rpt.PipelineTask.IsMatrixed() {
 			if err := resources.ValidateParameterTypesInMatrix(pipelineRunFacts.State); err != nil {
 				logger.Errorf("Failed to validate matrix %q with error %v", pr.Name, err)
 				pr.Status.MarkFailed(ReasonInvalidMatrixParameterTypes, err.Error())
 				return controller.NewPermanentError(err)
 			}
 		}
+	}
+
+	for _, rpt := range nextRpts {
```

Comment:
curious why this block is moving?

Reply:
it was leading to partial breakage in multiple tests. That was because we needed to move part of the parameter validation just before the creation of the task run. 
I therefore moved the matrix parameter validation in it's own block so that all matrix validation is completed before we start creating the task runs.

Reply:
I think that if I sync the workspaces and parameter propagation implementation first (ie. both happen in a similar way) then I might be able to undo this change. Let me do that. ie. Open a new issue and PR that syncs the implementation of propagated params and workspaces and then do the validation of referenced pipelines (this PR). WDYT?